File: algoexpert/python/medium/bfs.py
class Node:
    def __init__(self, name):
        self.children = []
        self.name = name

    def addChild(self, name):
        self.children.append(Node(name))
        return self

    def breadthFirstSearch(self, arr):
        queue = [self]
        while len(queue):
            node = queue.pop(0)
            arr.append(node.name)

            for child in node.children:
                queue.append(child)

        return arr

File: algoexpert/javascript/medium/bfs.js
class Node {
  constructor(name) {
    this.name = name;
    this.children = [];
  }

  addChild(name) {
    this.children.push(new Node(name));
    return this;
  }

  breadthFirstSearch(arr) {
    let queue = [this];

    while (queue.length) {
      let node = queue.shift();
      arr.push(node.name);

      for (let child of node.children) {
        queue.push(child);
      }
    }
    return arr;
  }
}
